packages/grid/x-grid/src/tests/reorder.XGrid.test.tsx

```diff
@@ -218,10 +227,151 @@ describe('<XGrid /> - Reorder', () => {
+      const columns = [
+        { field: 'brand', disableReorder: true },
+        { field: 'desc' },
+        { field: 'type' },
+      ];
+
+      const Test = () => {
+        apiRef = useGridApiRef();
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid apiRef={apiRef} rows={rows} columns={columns} onPageChange={() => {}} />
+          </div>
+        );
+      };
+
+      render(<Test />);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+      const dragCol = getColumnHeaderCell(1).firstChild!;
+      const targetCol = getColumnHeaderCell(0).firstChild!;
+
+      fireEvent.dragStart(dragCol);
+      fireEvent.dragEnter(targetCol);
+      const dragOverEvent = createDragOverEvent(targetCol);
+      fireEvent(targetCol, dragOverEvent);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+
+      const dragEndEvent = createDragEndEvent(dragCol);
+      fireEvent(dragCol, dragEndEvent);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+    });
+
+    it('should not allow to drag right of last visible column if it has disableReorder=true', () => {
+      let apiRef: GridApiRef;
+      const rows = [{ id: 0, brand: 'Nike' }];
+      const columns = [
+        { field: 'brand' },
+        { field: 'desc' },
+        { field: 'type', disableReorder: true },
+      ];
+
+      const Test = () => {
+        apiRef = useGridApiRef();
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid apiRef={apiRef} rows={rows} columns={columns} onPageChange={() => {}} />
+          </div>
+        );
+      };
+
+      render(<Test />);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+      const dragCol = getColumnHeaderCell(1).firstChild!;
+      const targetCol = getColumnHeaderCell(0).firstChild!;
+
+      fireEvent.dragStart(dragCol);
+      fireEvent.dragEnter(targetCol);
+      const dragOverEvent = createDragOverEvent(targetCol);
+      fireEvent(targetCol, dragOverEvent);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+
+      const dragEndEvent = createDragEndEvent(dragCol);
+      fireEvent(dragCol, dragEndEvent);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+    });
+
+    it('should allow to drag right of a column with disableReorder=true if it is not the last visible one', () => {
+      let apiRef: GridApiRef;
+      const rows = [{ id: 0, brand: 'Nike' }];
+      const columns = [
+        { field: 'brand' },
+        { field: 'desc', disableReorder: true },
+        { field: 'type' },
+      ];
+
+      const Test = () => {
+        apiRef = useGridApiRef();
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid apiRef={apiRef} rows={rows} columns={columns} onPageChange={() => {}} />
```

Comment:
```suggestion
            <XGrid apiRef={apiRef} rows={rows} columns={columns} />
```

x the new test case. I'm taking care of the old one in #2088

Reply:
> For the name of the property, I would refer to https://material-ui.com/guides/api/#property-naming

@oliviertassinari 

This page is for props, I don't think the shorthand argument really applies here.
But I am 100% in favor of consistency. 

Both approaches can make sense

- Always favor `default: false` (meaning that we should have `disableResize` instead of `resizable`)
or
- Always favor the affirmative form (meaning that we should have `exportable` instead of `disableExport`)

Most options are in the affirmative form so I think we should rename `disableExport` into `exportable` (my PR has not yet been merged) and on this PR `orderable` ?